src/schema/v2/me/__tests__/myCollectionUpdateArtworkMutation.test.ts

```diff
@@ -189,4 +201,146 @@ describe("myCollectionUpdateArtworkMutation", () => {
+
+      expect(createArtworkEditionSetLoader).toHaveBeenCalledWith(
+        updatedArtwork.id,
+        {
+          edition_size: null,
+          available_editions: ["50"],
+        }
+      )
+    })
+
+    it("works if you only specify the edition size", async () => {
+      const mutation = computeMutationInput({
+        editionSize: "50",
+      })
+
+      await runAuthenticatedQuery(mutation, defaultContext)
+
+      expect(createArtworkEditionSetLoader).toHaveBeenCalledWith(
+        updatedArtwork.id,
+        {
+          edition_size: "50",
+          available_editions: null,
+        }
+      )
+    })
+
+    it("does not create an edition set if you don't specify either", async () => {
+      const mutation = computeMutationInput({
+        editionNumber: null,
+        editionSize: null,
+      })
+
+      await runAuthenticatedQuery(mutation, defaultContext)
+
+      expect(createArtworkEditionSetLoader).not.toHaveBeenCalled()
+    })
+  })
+
+  describe("setting edition set info on an artwork with existing edition set info", () => {
+    const editionedArtwork = {
+      ...updatedArtwork,
+
+      edition_sets: [{ id: "my-edition-set-id" }],
+    }
+    beforeEach(() => {
+      updateArtworkLoader.mockResolvedValueOnce(editionedArtwork)
+    })
+
+    it("updates the edition set on the artwork", async () => {
+      const mutation = computeMutationInput({
+        editionNumber: "50",
+        editionSize: "100",
+      })
+
+      await runAuthenticatedQuery(mutation, defaultContext)
+
+      expect(updateArtworkEditionSetLoader).toHaveBeenCalledWith(
+        {
+          artworkId: updatedArtwork.id,
+          editionSetId: editionedArtwork.edition_sets[0].id,
+        },
+        {
+          edition_size: "100",
+          available_editions: ["50"],
+        }
+      )
+    })
+
+    it("works if you only specify the edition number", async () => {
+      const mutation = computeMutationInput({
+        editionNumber: "50",
+      })
+
+      await runAuthenticatedQuery(mutation, defaultContext)
+
+      expect(updateArtworkEditionSetLoader).toHaveBeenCalledWith(
+        {
+          artworkId: updatedArtwork.id,
+          editionSetId: editionedArtwork.edition_sets[0].id,
+        },
+        {
+          edition_size: null,
+          available_editions: ["50"],
+        }
+      )
+    })
+
+    it("works if you only specify the edition size", async () => {
+      const mutation = computeMutationInput({
+        editionSize: "50",
+      })
+
+      await runAuthenticatedQuery(mutation, defaultContext)
+
+      expect(updateArtworkEditionSetLoader).toHaveBeenCalledWith(
+        {
+          artworkId: updatedArtwork.id,
+          editionSetId: editionedArtwork.edition_sets[0].id,
+        },
+        {
+          edition_size: "50",
+          available_editions: null,
+        }
+      )
+    })
+
+    it("does not create an edition set if you don't specify either", async () => {
```

Comment:
[not-sure-if-blocking]  

Is this the right thing to do if there's no editionNumber/editionSize on edit?

I'm worried it will translate to an experience where I edit an artwork in my collection and remove the edition number/size, but the system doesn't actually remove it when I save.

Maybe there are other places you're anticipating handling that scenario?

Reply:
Very good catch!!!! will fix tomorrow